File: src/libs/cpp/functions/frombytes.cpp
#include "frombytes.hpp"

void frombytes(Context &context) noexcept
{
	// Convert a list of bytes into a number
	auto bytes = std::get<std::vector<Value>>(context.stack.pop());
	int result = 0;
	for (const Value &byte : bytes)
	{
		result *= 256;
		result += static_cast<int>(byte.to_number()) % 256;
	}
	context.stack.push(result);
}

File: src/runtime/functions/frombytes.lua
return function(vm)
	local v = vm.pop()
	if type(v[1]) ~= 'table' then
		vm.push(0)
	else
		local result = 0
		for i = 1, #v[1] do
			result = result * 256 + v[1][i]
		end
		vm.push(result)
	end
end
